cli/permissions.rs

```diff
@@ -170,51 +180,123 @@ impl DenoPermissions {
   pub fn check_read(&self, filename: &str) -> DenoResult<()> {
     match self.allow_read.get_state() {
       PermissionAccessorState::Allow => Ok(()),
-      PermissionAccessorState::Ask => match self
-        .try_permissions_prompt(&format!("read access to \"{}\"", filename))
-      {
-        Err(e) => Err(e),
-        Ok(v) => {
-          self.allow_read.update_with_prompt_result(&v);
-          v.check()?;
+      state => {
+        if check_path_white_list(filename, &self.read_whitelist) {
           Ok(())
+        } else {
+          match state {
+            PermissionAccessorState::Ask => match self.try_permissions_prompt(
+              &format!("read access to \"{}\"", filename),
+            ) {
+              Err(e) => Err(e),
+              Ok(v) => {
+                self.allow_read.update_with_prompt_result(&v);
+                v.check()?;
+                Ok(())
+              }
+            },
+            PermissionAccessorState::Deny => Err(permission_denied()),
+            _ => unreachable!(),
+          }
         }
-      },
-      PermissionAccessorState::Deny => Err(permission_denied()),
+      }
     }
   }
 
   pub fn check_write(&self, filename: &str) -> DenoResult<()> {
     match self.allow_write.get_state() {
       PermissionAccessorState::Allow => Ok(()),
-      PermissionAccessorState::Ask => match self
-        .try_permissions_prompt(&format!("write access to \"{}\"", filename))
-      {
-        Err(e) => Err(e),
-        Ok(v) => {
-          self.allow_write.update_with_prompt_result(&v);
-          v.check()?;
+      state => {
+        if check_path_white_list(filename, &self.write_whitelist) {
           Ok(())
+        } else {
+          match state {
+            PermissionAccessorState::Ask => match self.try_permissions_prompt(
+              &format!("write access to \"{}\"", filename),
+            ) {
+              Err(e) => Err(e),
+              Ok(v) => {
+                self.allow_write.update_with_prompt_result(&v);
+                v.check()?;
+                Ok(())
+              }
+            },
+            PermissionAccessorState::Deny => Err(permission_denied()),
+            _ => unreachable!(),
+          }
         }
-      },
-      PermissionAccessorState::Deny => Err(permission_denied()),
+      }
     }
   }
 
-  pub fn check_net(&self, domain_name: &str) -> DenoResult<()> {
+  pub fn check_net(&self, url_or_host_and_port: &str) -> DenoResult<()> {
```

Comment:
Why not just host_and_port?

Reply:
URLs and host and port strings(like `localhost:4545` or `localhost`) are both valid.

Reply:
But supporting URLs means that you have to parse the URL here. Whereas it could be parsed outside of this and the host string extracted.

Reply:
I would be satisfied if there were two methods check_net() and check_net_url()